### PR TITLE
Analyzer statistics

### LINKED ISSUE #565: Store system statistics
It could be useful if, for an analysis, we stored which version of clang and clang-tidy was used for analysis and various other metadata (discuss?).

Also if we are already modifying the database and interface to accompany the storage of this metadata, showing the "version" of CodeChecker on the web interface and in the help command would also be nice.

### PR DESCRIPTION
**Pull request by csordasmarton**
> Closes #565

* Add CodeChecker version to the run history table and shows this at the Run and Run history pages.
* Create an analyzer statistic table which stores the number of failed and successfully analyzed files for each analyzer types.
* Show analyzer statistics at the run history and run pages.

### CODE REVIEW

**Comment by bruntib** [comment]
What is the granularity of the DB schema update processes? Shouldn't this and "bug path length" column be introduced in one step? Is there any difference in the performance of schema upgrade?

**Comment by dkrupp** [comment]
Re-testing it with re-analyzing too the feature seems to work now, however in the runs table the "CodeChecker version" column is only shown for the first run.

File: libcodechecker/cmd/cmd_line_client.py [hunk 1]
```diff
@@ -235,17 +235,29 @@ def handle_list_runs(args):
         print(CmdLineOutputEncoder().encode(results))
 
     else:  # plaintext, csv
-        header = ['Name', 'Number of reports', 'Storage date', 'Version tag',
-                  'Duration']
+        header = ['Name', 'Number of unresolved reports',
+                  'Analyzer statistics', 'Storage date', 'Version tag',
+                  'Duration', 'Client version']
```
Comment (gyorb) [inline libcodechecker/cmd/cmd_line_client.py]: Please be consistent with the web UI where `CodeChecker version` is used.

File: libcodechecker/cmd/cmd_line_client.py [hunk 2]
```diff
@@ -1178,12 +1190,23 @@ def handle_list_run_histories(args):
     if args.output_format == 'json':
         print(CmdLineOutputEncoder().encode(run_history))
     else:  # plaintext, csv
-        header = ['Date', 'Run name', 'Version tag', 'User']
+        header = ['Date', 'Run name', 'Version tag', 'User',
+                  'Client version', 'Analyzer statistics']
```
Comment (gyorb) [inline libcodechecker/cmd/cmd_line_client.py]: Please be consistent with the web UI where CodeChecker version is used.

File: www/changelog.html [hunk 3]
```diff
@@ -92,6 +92,22 @@ <h3>Support managing source component on the GUI</h3>
       </div>
     </div>
 
+    <div class="intro-row">
+      <h3>Analysis Statistics</h3>
+      <div class="content">
+        <div class="intro-col intro-col-left col-md-2">
+          It is displayed for each run, which CodeChecker and Clang Analyzer
```
Comment (gyorb) [inline www/changelog.html]: I would rephrase the short description a bit:
`For each run the CodeChecker version (used to run the analysis) and the static analyzer versions along with the success/failure rate of the analysis can be viewed.`

File: www/scripts/codecheckerviewer/AnalyzerStatisticsDialog.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          dom.create('b', { innerHTML : 'Version: ' }, li);
+
+          var versionList = dom.create('ul', { class : 'version' }, li);
+          stats[analyzer].version.trim().split('\n').forEach(function (item) {
+            dom.create('li', { innerHTML : item.trim() }, versionList);
+          });
+        }
+
+        // Successfully analyzed files.
+        var li = dom.create('li', {
```
[the remark this answers is not in the document]
Comment (gyorb) [reply www/scripts/codecheckerviewer/AnalyzerStatisticsDialog.js]: Could you rename the `li` variables to something different so no re declaration is done.
